components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts

```diff
@@ -1,78 +1,64 @@
 import {
-  Component, EventEmitter, OnInit, Input, Output
+  Component, EventEmitter, Input, Output, OnChanges
 } from '@angular/core';
 
-import { orderBy } from 'lodash/fp';
-
-import { Project, ProjectConstants } from 'app/entities/projects/project.model';
+import { ProjectConstants, Project } from 'app/entities/projects/project.model';
 
 const { UNASSIGNED_PROJECT_ID } = ProjectConstants;
 
+// Extend the project model with the checked field.
+// This represents whether the project's checkbox is unchecked or not
+// in this component's UI
+export interface ProjectChecked extends Project {
+  checked: boolean;
+}
+
 @Component({
   selector: 'app-projects-dropdown',
   templateUrl: './projects-dropdown.component.html',
   styleUrls: ['./projects-dropdown.component.scss']
 })
-export class ProjectsDropdownComponent implements OnInit {
-
-  // The array of projects that you pass into the component. This array gets updated when a user
-  // makes changes and is emitted to the parent component.
-  @Input() projects:  Array<Project> = [];
-
-  // Setting required to true means the dropdown will show an error if a user tries to close the
-  // dropdown but hasn't selected any projects.
-  @Input() required = false;
+export class ProjectsDropdownComponent implements OnChanges {
+  // The map of ProjectChecked by id. Any checked changes propagated via
+  // onProjectChecked. Updates should be applied to parent component state.
+  @Input() projects: { [id: string]: ProjectChecked } = {};
```

Comment:
Another spot to use the `ProjectMap` type, so this would become `public projects: ProjectMap = {};`